lib/Dancer2/Core/App.pm

```diff
@@ -1619,6 +1620,33 @@ sub response_not_found {
     return $response;
 }
 
+sub uri_for_route {
+    my ( $self, $route_name, $named_params, $unnamed_params, $dont_escape ) = @_;
+
+    my $route =
+      List::Util::first { $_->name eq $route_name } @{$self->routes->{'get'}};
+
+    my $string = $route->spec_route;
+
+    $string =~ /[:@]/
+        or return $string;
+
+    # XXX This regex here are taken from
+    #     Dancer2::Core::Route::_build_regexp_from_string.
+    #     Would anyone like to refactor? -- SX.
+    my @params = $string =~ m{:([^/\.\?]+)}g;
+    foreach my $param (@params) {
+        my $value = $named_params->{$param}
+            or Carp::croak("$param is not a part of route $route_name");
```

Comment:
This error message confuses me --- we know `$param` is in the route, since it came from `$route->spec_route`.  I think this should be

    Route $route_name uses parameter ${param}, which was not provided

or some such.  Am I missing something?

Reply:
We'll change this. Thanks for pointing it out.

Reply:
👍

Reply:
Resolved